### ISSUE TITLE
Auto-increment of Primary key does not work

### ISSUE THREAD

**Issue by arashlayeghi** [hasura/graphql-engine#5737]
We are using the latest stable version of Hasura v.1.3.1

We have a table with a primary key of type Integer(Auto-Increment):
![image](https://user-images.githubusercontent.com/22353246/92370446-8bc57b00-f10f-11ea-81da-6fa41bfa6212.png)

However, when I was writing a script to reset the database to mock data I passed the primary key IDs in the insert mutation so that the PKs and the FKs to it are always fixed. Now I see that the auto-increment of the primary key does not work and I get `Uniqueness violation. duplicate key value violates unique constraint \"location_pkey\"` error when I don't pass the `id` in the mutation and even worst I cannot insert a row in the Insert Row tab. 

![image](https://user-images.githubusercontent.com/22353246/92370811-0a221d00-f110-11ea-8a9f-4a26d0d589aa.png)

![image](https://user-images.githubusercontent.com/22353246/92370873-21610a80-f110-11ea-82fa-f8a25b165ce0.png)

It seems after I manually set the IDs in the insert mutation now the auto-increment mechanism got confused and doesn't work anymore. How can I have it again for my tables?

**Comment by tirumaraiselvan** [comment]
You might want to "restart" the current value of the sequence `shared.location_id_seq` : https://www.postgresql.org/docs/current/sql-altersequence.html

**Comment by Veshu-Arora** [comment]
Hey @arashlayeghi, i would like to work on this issue. And i would like to have some more acces...
1. I would like to have access to the table you are storing your data into with permission to perform CRUD using test data either directly or using an interface or some UI.

**Comment by Veshu-Arora** [comment]
Hey @arashlayeghi, you see, whenever you insert data into the table using an application , you dont need to pass the id because the database itself inserts the id and the application(or may be the database) also remembers what was the  id of the last row which was inserted using the application so that it assigns next id just 1 greater than the id that was assigned to last row. Say that the last row i inserted using application was assigned an id of 4 ,  and then i inserted 6 rows manually, giving them the ids 5,6,7,8,9 and 10. Now , if i will try to insert a row using the application, the database will try to assign this row an id = 5 because it remembers that the last row which this application inserted was given an id = 4 and it will give the error that the unique constraint is violated because it now has a row with id = 5 which you had inserted. 
So , either you can delete all rows you inserted manually and the error should go away and then insert the rows using the application and not manually.
Or, In case you do not delete the manually inserted rows, you should get this error as many times as the number of rows you inserted manually( 6 times in case of example i gave above) and then the the error will go away because the 7th time you insert the row using the application, the database will assign the id = 11 which does not exist in table and your problem would be solved.
But, deleting the manually inserted rows would be better choice.

**Comment by arashlayeghi** [comment]
Hey, @Veshu-Arora Thanks for your explanation. 
I appreciated what you said and realized the root cause of the problem. However, it's not the solution to our problem. 
In our use-case, we need to manually insert id for some rows which are mock data. Also, we need that after that we can insert some new rows without id with auto-increment. We cannot simply delete those manually inserted rows to solve this issue. 

After all, this is Hasura which is managing PostgreSQL for us, and when we explicitly set the ids in insert mutation it is the responsibility of Hasura to update the maximum id.

**Comment by joshuarobs** [comment]
Getting this same error. I can't insert any new rows without this error in the hasura console (and in my app if I call the graphql code). There should be a button that you can press within the hasura console to fix issues like this, to keep things in sync

**Comment by Shalashtein** [comment]
Same error here, it shouldn't be difficult for Hasura to check the max id and generate an id that's one more than that?

**Comment by AaravShah042** [comment]
Yeah, because it is possible that sometimes we have to add the record manually and after that, we want to add using mutation
Example:-
Report table

Some are default reports added by developer(We are adding manually) and user can create custom reports for that we have to call mutation
